File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/receivers/BootReceiver.java
package com.sorcererxw.matthiasheiderichphotography.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.sorcererxw.matthiasheiderichphotography.MHApp;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/17
 */

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        MHApp.getInstance().getScheduleManager().startAutoChangeWallpaperIfNeeded();
    }
}

File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/receivers/BootReceiver.kt
package com.sorcererxw.matthiasheiderichphotography.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.sorcererxw.matthiasheiderichphotography.MHApp

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/17
 */

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        MHApp.instance!!.scheduleManager!!.startAutoChangeWallpaperIfNeeded()
    }
}